### PR TITLE
[regex] Add 3 new Regex workouts

### PR DESCRIPTION
**Pull request by emmbow**

### CODE REVIEW

**Comment by enki-hq** [comment]
Failed parsing files: 
  - regular-expressions/introduction/practical-applications-ii/credit-cards-i.md: 
 > 	Cannot parse invalid footnotes

**Comment by enki-hq** [comment]
Failed parsing files: 
  - regular-expressions/introduction/practical-applications-ii/credit-cards-i.md: 
 > 	Cannot parse invalid footnotes

**Comment by enki-hq** [comment]
Failed parsing files: 
  - regular-expressions/introduction/regex-use-cases-ii/credit-cards-i.md: 
 > 	Cannot parse invalid footnotes

**Comment by enki-hq** [comment]
Failed parsing files: 
  - regular-expressions/introduction/regex-use-cases-ii/credit-cards-i.md: 
 > 	Cannot parse invalid footnotes

File: regular-expressions/introduction/regex-use-cases-ii/credit-cards-i.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+- [-\s]?
+- [-\s]
+- ?
+- \d
+
+---
+## Footnotes
+
+[1. Credit Card Networks]
```
Comment (kapnobatai137) [inline regular-expressions/introduction/regex-use-cases-ii/credit-cards-i.md:81]: should be `:` instead of `.`
```suggestion
[1: Credit Card Networks]
```

**Review by kapnobatai137 — CHANGES_REQUESTED** [review]

**Comment by enki-hq** [comment]
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| html-tags | Insight slug `html-tags` is not unique in the database. |
### Title

| File Name | Error Message |
|:-:|:-:|
| regex-use-cases-ii | Workout title should be `` Regex Use Cases Ii `` |
| regex-use-cases-iii | Workout title should be `` Regex Use Cases Iii `` |
| the-regexp-object | Insight title should be `` The Regexp Object `` |
| regex-email-address-checker-ii | Insight title should be `` Email Address Checker Ii `` |
| regex-password-validation-using-positive-lookaheads | Insight title should be `` Passwords Using Positive Lookaheads `` |
| credit-cards-ii | Insight title should be `` Credit Cards Ii `` |
| url | Insight title should be `` Urls `` |
| zip-codes | Insight title should be `` Zip Codes `` |
| html-tags | Insight title should be `` HTML Tags `` |

> If you notice anything wrong with these rules contact andrei@enki.com.

**Comment by kapnobatai137** [comment]
@emmbow if the title suggestions look weird (like saying `II` should be `Ii`), ignore them. there is a fix in the pipeline

**Comment by enki-hq** [comment]
### Title

| File Name | Error Message |
|:-:|:-:|
| regex-use-cases-ii | Workout title should be `` Regex Use Cases Ii `` |
| regex-use-cases-iii | Workout title should be `` Regex Use Cases Iii `` |
| the-regexp-object | Insight title should be `` The Regexp Object `` |
| regex-email-address-checker-ii | Insight title should be `` Email Address Checker Ii `` |
| regex-password-validation-using-positive-lookaheads | Insight title should be `` Passwords Using Positive Lookaheads `` |
| credit-cards-ii | Insight title should be `` Credit Cards Ii `` |
| url | Insight title should be `` Urls `` |
| zip-codes | Insight title should be `` Zip Codes `` |

> If you notice anything wrong with these rules contact andrei@enki.com.

File: regular-expressions/introduction/regex-and-javascript/exec.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+tags:
+  - introduction
+
+type: normal
+
+category: must-know
+
+links:
+	- '[.exec()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec){website}'
```
Comment (kapnobatai137) [inline regular-expressions/introduction/regex-and-javascript/exec.md:12]:
```suggestion
  - '[.exec()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec){website}'
```

File: regular-expressions/introduction/regex-and-javascript/test.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+tags:
+  - introduction
+
+type: normal
+
+category: must-know
+
+links: 
+	- '[.test()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test){website}'
```
Comment (kapnobatai137) [inline regular-expressions/introduction/regex-and-javascript/test.md:12]:
```suggestion
  - '[.test()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test){website}'
```

**Review by kapnobatai137 — CHANGES_REQUESTED** [review]

**Comment by enki-hq** [comment]
### Title

| File Name | Error Message |
|:-:|:-:|
| regex-use-cases-ii | Workout title should be `` Regex Use Cases Ii `` |
| regex-use-cases-iii | Workout title should be `` Regex Use Cases Iii `` |
| the-regexp-object | Insight title should be `` The Regexp Object `` |
| regex-email-address-checker-ii | Insight title should be `` Email Address Checker Ii `` |
| regex-password-validation-using-positive-lookaheads | Insight title should be `` Passwords Using Positive Lookaheads `` |
| credit-cards-ii | Insight title should be `` Credit Cards Ii `` |
| url | Insight title should be `` Urls `` |
| zip-codes | Insight title should be `` Zip Codes `` |

> If you notice anything wrong with these rules contact andrei@enki.com.

**Comment by enki-hq** [comment]
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| html-tags | Insight slug `html-tags` is not unique in the database. |
### Title

| File Name | Error Message |
|:-:|:-:|
| regex-email-address-checker-ii | Insight title should be `` Email Address Checker Ii `` |
| regex-password-validation-using-positive-lookaheads | Insight title should be `` Passwords Using Positive Lookaheads `` |
| credit-cards-ii | Insight title should be `` Credit Cards Ii `` |
| url | Insight title should be `` Urls `` |
| zip-codes | Insight title should be `` Zip Codes `` |
| regex-basics | Insight title should be `` Regex Basics `` |
| regex-quantifiers-ii | Insight title should be `` Quantifiers Ii `` |
| regex-wildcards-ii | Insight title should be `` Wildcards Ii `` |
| regex-ranges-ii | Insight title should be `` Ranges Ii `` |
| the-regexp-object | Insight title should be `` The Regexp Object `` |

> If you notice anything wrong with these rules contact andrei@enki.com.

**Comment by enki-hq** [comment]
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| html-tags | Insight slug `html-tags` is not unique in the database. |
### Title

| File Name | Error Message |
|:-:|:-:|
| regex-email-address-checker-ii | Insight title should be `` Email Address Checker Ii `` |
| regex-password-validation-using-positive-lookaheads | Insight title should be `` Passwords Using Positive Lookaheads `` |
| credit-cards-ii | Insight title should be `` Credit Cards Ii `` |
| url | Insight title should be `` Urls `` |
| zip-codes | Insight title should be `` Zip Codes `` |
| regex-basics | Insight title should be `` Regex Basics `` |
| regex-quantifiers-ii | Insight title should be `` Quantifiers Ii `` |
| regex-wildcards-ii | Insight title should be `` Wildcards Ii `` |
| regex-ranges-ii | Insight title should be `` Ranges Ii `` |
| the-regexp-object | Insight title should be `` The Regexp Object `` |

> If you notice anything wrong with these rules contact andrei@enki.com.

**Comment by enki-hq** [comment]
### Title

| File Name | Error Message |
|:-:|:-:|
| regex-email-address-checker-ii | Insight title should be `` Email Address Checker Ii `` |
| regex-password-validation-using-positive-lookaheads | Insight title should be `` Passwords Using Positive Lookaheads `` |
| credit-cards-ii | Insight title should be `` Credit Cards Ii `` |
| url | Insight title should be `` Urls `` |
| zip-codes | Insight title should be `` Zip Codes `` |
| regex-basics | Insight title should be `` Regex Basics `` |
| regex-quantifiers-ii | Insight title should be `` Quantifiers Ii `` |
| regex-wildcards-ii | Insight title should be `` Wildcards Ii `` |
| regex-ranges-ii | Insight title should be `` Ranges Ii `` |
| the-regexp-object | Insight title should be `` The Regexp Object `` |

> If you notice anything wrong with these rules contact andrei@enki.com.

**Review by kapnobatai137 — APPROVED** [review]

**Pull request merged**